### DISCUSSION TITLE
Assign an initial value to state variables

### DISCUSSION THREAD

**Discussion by ferhannb** [casadi/casadi#3726] [Q&A]
Hello 

I have a set of equations that I specified in the code below. 

        # States 
        delta_theta = ca.SX.sym('delta_theta')
        delta_X        = ca.SX.sym('delta_X')
        delta_Y        = ca.SX.sym('delta_Y')
        theta           = ca.SX.sym('theta')
        X                 =ca.SX.sym('X')
        Y                 = ca.SX.sym('Y')
        states         =ca.vertcat(delta_theta,delta_X,delta_Y,theta,X,Y)
        number_states = states.numel()

        # Inputs
        K           = ca.SX.sym('K')
        delta_s = ca.SX.sym('delta_s')
        inputs  =ca.vertcat(K,delta_s)
        number_inputs = inputs.numel()
        
        system_eq = ca.vertcat( K* delta_s,
                                     ca.if_else(K!=0,(ca.sin(theta)-ca.sin(theta-delta_theta))/K,ca.cos(theta)*delta_s),
                                     ca.if_else(K!=0,-(ca.cos(theta)-ca.cos(theta-delta_theta))/K,ca.sin(theta)*delta_s),    
                                     theta+delta_theta,
                                     X+delta_X,
                                     Y+delta_Y       
                                    )
        
        f = ca.Function('f',[states,inputs],[system_eq])
`
I get the error "Cannot create 'solver' since [X, Y] are free." for the statuses X and Y in the *system_eq* variable representing the equation of the system. To resolve this error, X and Y must take an initial value, how can I assign an initial value to the statuses X and Y?

**Comment by b-mare** [comment]
Hi,

On which line do you encounter your error? It seems to me there is nothing wrong with the snippet you provided...

The error suggests that you created a solver object without specifying what X and Y are for the solver.
Something like this for example gives the error you mentioned:

```
nlp = {'x':ca.vertcat(delta_theta, delta_X, delta_Y, theta),  # Lacks the states X and Y
       'p': inputs,
       'g':system_eq}
S = ca.nlpsol('S', 'ipopt', nlp)
```
